mlrun/runtimes/nuclio/api_gateway.py

```diff
@@ -134,21 +295,21 @@ def __init__(
         :param host:  The host of the API gateway (optional). If not set, it will be automatically generated
         :param canary: The canary percents for the API gateway of type list[int]; for instance: [20,80]
         """
-        self.functions = None
-        self._validate(
+        self.metadata = APIGatewayMetadata(
+            name=name,
             project=project,
+        )
+
+        self.spec = APIGatewaySpec(
```

Comment:
see the following example on how to define meta and spec. constructor should only have meta and spec params
```
    @property
    def metadata(self) -> ProjectMetadata:
        return self._metadata

    @metadata.setter
    def metadata(self, metadata):
        self._metadata = self._verify_dict(metadata, "metadata", ProjectMetadata)

    @property
    def spec(self) -> ProjectSpec:
        return self._spec

    @spec.setter
    def spec(self, spec):
        self._spec = self._verify_dict(spec, "spec", ProjectSpec)
```